src/app/shared/state/search/search.action.ts

```diff
@@ -0,0 +1,29 @@
+export class SetGenderFilter {
+  static readonly type = '[Search API] Set Gender Filter';
+  constructor(readonly gender: 'male' | 'female' | 'male-female') { }
```

Comment:
I wonder if we should just have 'male' | 'female' | undefined? Then it would act like the others in that if there is nothing set, it doesn't limit the search.